src/components/button-swap/button-swap.test.ts

```diff
@@ -0,0 +1,177 @@
+import path from 'path';
+import type { PleasantestUtils } from 'pleasantest';
+import { withBrowser } from 'pleasantest';
+import { loadTwigTemplate, loadGlobalCSS } from '../../../test-utils';
+
+// Helper to load the Twig template file
+const componentMarkup = loadTwigTemplate(
+  path.join(__dirname, './button-swap.twig')
+);
+
+// Helper to initialize the component JS
+const initJS = (
+  utils: PleasantestUtils,
+  initialCallback?: () => void,
+  swappedCallback?: () => void
+) =>
+  utils.runJS(
+    `
+    import { initButtonSwap } from './button-swap'
+    export default (initialCallback, swappedCallback) => {
+      initButtonSwap(document.querySelector('.js-button-swap'), {
+        initialCallback,
+        swappedCallback
+      })
+    }
+    `,
+    [initialCallback, swappedCallback]
+  );
+
+test(
+  'Initial state',
+  withBrowser(async ({ utils, screen }) => {
+    await utils.injectHTML(await componentMarkup());
+    await loadGlobalCSS(utils);
+
+    // Visually hidden text for a more inclusive UX
+    const statusMsg = await screen.getByRole('status');
+    await expect(statusMsg).toHaveTextContent(
+      /^currently unsubscribed from notifications$/i
+    );
+
+    const statusMsgs = await screen.getAllByRole('status');
+    expect(statusMsgs.length).toBe(1);
+
+    const firstBtn = await screen.getByRole('button', {
+      name: /^get notifications$/i,
+    });
+    await expect(firstBtn).toBeVisible();
+
+    const secondBtn = await screen.queryByRole('button', {
+      name: /^turn off notifications$/i,
+    });
+    expect(secondBtn).toBeNull();
+  })
+);
+
+test(
+  'Swap UI state when clicked',
+  withBrowser(async ({ utils, screen, user }) => {
+    await utils.injectHTML(await componentMarkup());
+    await loadGlobalCSS(utils);
+    // I'd like to avoid using a test ID, but I couldn't figure out a different way.
+    // @todo Can this be done without at test ID?
```

Comment:
Is this comment still relevant?

Reply:
Oops, great catch! This comment is no longer relevant. 🙂

Reply:
Deleted! https://github.com/cloudfour/cloudfour.com-patterns/pull/1516/commits/a64e04b59a7b0ca0768d68d02661cb6d653ebd93